pdfgen.c

```diff
@@ -2622,20 +2760,64 @@ static int pdf_add_png_data(struct pdf_doc *pdf, struct pdf_object *page,
         goto info_free;
     }
 
-    /* RGB colored image */
+    char *colour_space = NULL;
```

Comment:
As you mentioned, string handling in C sort of sucks. I've got the `dstr` stuff in here in an attempt to mitigate it. So you can use:
```c
struct dstr colour_space = INIT_DSTR;
dstr_append(&colour_space, "/DeviceGray");
...
dstr_free(&colour_space)
```
If the string is < 64 bytes long, no dynamic allocation will actually occur (it'll just live on the stack). If it ends up being longer, it'll extend as needed

Reply:
thank you for the info, I've implemented it like this now.